packages/wallet/src/account.test.ts

```diff
@@ -13,10 +13,23 @@ import type {
 } from '@fuel-ts/providers';
 import { Provider } from '@fuel-ts/providers';
 import * as providersMod from '@fuel-ts/providers';
-import * as transactionReqMod from '@fuel-ts/providers/src/transaction-request/transaction-request';
 
 import { Account } from './account';
 
+// TODO: Check if there's a better alternative to this
+/**
+ * This makes it possible to mock modules that are exported
+ * from package's index files, using exports syntax such as:
+ *
+ *  export * from '...'
+ *
+ * https://stackoverflow.com/a/72885576
+ */
+jest.mock('@fuel-ts/providers', () => ({
+  __esModule: true,
+  ...jest.requireActual('@fuel-ts/providers'),
+}));
```

Comment:
The previous code imported an internal file from the `provider` package **_directly_** (which should not be allowed) to mock it before running the tests. The direct import was only possible because of some _[tiny]_ conflicts between the usage of the legacy `main` field combined with the `exports` one — _when these are pointing to `*.ts` files directly_.

After removing the legacy properties, importing inner modules from packages is not possible anymore. This is to keep things tight and coherent, but the _fix_ bumped into another (ESM / Jest?) issue that required the hack above to keep the mocking step working for these tests.

In short, this workaround _proxies the imported package while injecting the `__esModule: true` property, making it possible for properties to be **redefined** (mocked) in runtime_.